datalad/local/add_readme.py

```diff
@@ -82,6 +90,25 @@ def __call__(dataset, filename='README.md', existing='skip'):
         # unlock, file could be annexed
         if lexists(filename):
             dataset.unlock(filename)
+        if not lexists(filename):
+            if isinstance(dataset.repo, AnnexRepo):
+                # if we have an annex repo, shall the README go to Git or annex?
+                if dataset.status(
+                        '.gitattributes',
+                        result_renderer='disabled',
+                        return_type='list',)[0]['state'] != 'clean':
+                    raise RuntimeError(
+                        'Stopping, because the .gitattributes file is modified'
+                        )
+                # unless --annex is set, configure the README to go into Git
+                dataset.repo.set_gitattributes(
+                    [('README.md', {'annex.largefiles': 'anything' if annex
```

Comment:
Well, you want to do this even if .gitattributes doesn't exist - datalad does create it to force md5e backend, but a pure annex repo might not have it

Reply:
Overall - why to condition on. gitattributes existence? We can have it or not, but we want to enforce this fine too go to git or annex

Reply:
> Overall - why to condition on. gitattributes existence? We can have it or not, but we want to enforce this fine too go to git or annex

mhh, maybe I misunderstood what ``set_gitattributes`` does - does it add a condition on the existence of this file? I thought it rather creates one if it can't find it?

Reply:
(or maybe I misunderstood your second comment?)

Reply:
Missed that. The issue is, that `status` on a not existing path would return an empty list. That's an indirect conditioning, since the call will fail with `index out of range` (your `[0]`). Don't see another kind of conditioning here.

Reply:
Ah, I see, thanks! But some check would need to be in place to make sure that there are no uncommitted changes in gitattributes before applying and saving the configuration for the README, right?

Reply:
So, safeguard needed. First get the status, and then check it only if something was returned. Otherwise there is no `.gitattributes` and you can safely add as well.

On that note another nitpick:
`['state'] != 'clean'` doesn't exactly imply "modified". Could be untracked. Probabyl to the same consequence - we don't exactly know whether it's ok to add something.

Reply:
> But some check would need to be in place to make sure that there are no uncommitted changes in gitattributes before applying and saving the configuration for the README, right?

Right. Just get the status in variable and do the check on `clean` only if there was a result. If not - all good, go ahead and write to it.

Reply:
thanks, this was helpful! I have pushed a commit that should implement it. If a .gitattributes file does not exist, it will be created